### PR TITLE
docs: add adr for disabling kiali operator

### PR DESCRIPTION
**Pull request by MxNxPx**
addresses https://github.com/defenseunicorns/uds-package-dubbd/issues/491

### CODE REVIEW

**Comment by corang** [comment]
Is skipping the operator and deploying kiali directly an option here? I think it still provides value

**Comment by MxNxPx** [comment]
> Is skipping the operator and deploying kiali directly an option here? I think it still provides value

@corang - i hope that i am wrong... but i think @mjnagel had said BB doesn't have straight Kiali available

**Comment by mjnagel** [comment]
@MxNxPx @corang BB does not publish a Kiali only chart, but the image would be the same one from Ironbank. It would mean we have to maintain our own chart with the BB integration pieces (or some 🌶️ magic to do that). Not totally infeasible but not as easy as adding a new gitrepo in our values.

File: docs/adrs/0003-disable-kiali-operator-for-dubbd.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+Accepted
+
+## Context
+
+Big Bang core (umbrella) is configured with certain "defaults" such as charts enabled or disabled and specific application configurations. It is preferred that to stay aligned wherever possible with these defaults, but in certain scenarios DUBBD may deviate from them, for example:
+
+* Simplification
+* Performance optimization
+* Compliance / controls alignment
+* Best practices (from delivery experience)
```
Comment (mjnagel) [inline docs/adrs/0003-disable-kiali-operator-for-dubbd.md:16]: Recommend we remove "delivery" wording (here and a few other spots below) and switch to user/customer to use more public facing language here.

**Comment by blancharda** [comment]
Some end users have expressed concern over (potentially) losing access to the tool.

They described it as a "single pane of glass" view for monitoring the current state of their deployments and in-cluster communication, which they use regularly during upgrade cycles to monitor health.

My preference would be to leave it in place until we have an alternative recommendation for installing it ad-hoc.
(And perhaps doing a broader user survey to better understand more end-user's usage patterns.)

That said, the underlying resource usage problem is still a problem..
+1 for @corang's suggestion above.. the [kiali-server chart](https://kiali.io/docs/installation/installation-guide/install-with-helm/) may be a potential starting point.

File: docs/adrs/0003-disable-kiali-operator-for-dubbd.md (hunk 1 above)
Comment (MxNxPx) [reply docs/adrs/0003-disable-kiali-operator-for-dubbd.md]: good call @mjnagel - replaced with the more general "platform" verbiage via [this commit](https://github.com/defenseunicorns/uds-package-dubbd/commit/50d6fcaf5e68c199de85eb1033101feda8f0825d)

**Review by corang — CHANGES_REQUESTED** [review]
Leaving it in is fine, but we still haven't addressed the problem. Kiali is still of no value if it never gets deployed by the operator

**Comment by corang** [comment]
> Leaving it in is fine, but we still haven't addressed the problem. Kiali is still of no value if it never gets deployed by the operator

It's probably the most work out of all the issues but it seems like deploying kiali ourself directly without the operator solves the most problems.

**Comment by corang** [comment]
> > Leaving it in is fine, but we still haven't addressed the problem. Kiali is still of no value if it never gets deployed by the operator
> 
> It's probably the most work out of all the issues but it seems like deploying kiali ourself directly without the operator solves the most problems.

Or as a bandaid just bump the memory limit a few gigs for now and mark it as a known issue

**Comment by MxNxPx** [comment]
> > > Leaving it in is fine, but we still haven't addressed the problem. Kiali is still of no value if it never gets deployed by the operator
> > 
> > 
> > It's probably the most work out of all the issues but it seems like deploying kiali ourself directly without the operator solves the most problems.
> 
> Or as a bandaid just bump the memory limit a few gigs for now and mark it as a known issue

@corang - agreed.  i have opened [this ticket](https://github.com/defenseunicorns/uds-package-dubbd/issues/505) to address separately from this ADR.

**Review by mjnagel — APPROVED** [review]

**Review by corang — APPROVED** [review]
Given the other ticket this is fine

**Pull request merged**